src/java/org/apache/cassandra/auth/IInternodeAuthenticator.java

```diff
@@ -82,11 +82,18 @@ default void setupInternode()
     }
 
     /**
-     * Enum that represents connection type of an internode connection.
+     * Enum that represents connection type of internode connection.
+     *
+     * INBOUND - called after connection established, with certificate available if present.
+     * OUTBOUND - called after connection established, with certificate available if present.
+     * REDIRECT - called before initiating a redirected connection, without certificate available.
+     * The outbound connection will be authenticated with the certificate once a redirected connection is established.
+     * This is an extra check can be used to detect misconfiguration before reconnection, or ignored by returning true.
```

Comment:
This is an extra check can be used to detect misconfiguration -> This is an extra check *that* can be used to detect misconfiguration